fitbenchmarking/fitbenchmarking
# Fixes table figure in concept docs

#### Description of Work

Fixes #626 

Updates table figure and adds the data used to create the figure in the docs.

Also, this PR fixes the Jacobian evaluation for the cost function.

#### Testing Instructions

1. run `fitbenchmarking -o docs/concept_figure/concept.ini docs/concept_figure/data` and check the results are consistent with the concept docs
2. check `Jacobian.eval_r_norm` is implemented correctly

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
